test/amoc_config_attributes_SUITE.erl

```diff
@@ -0,0 +1,199 @@
+
+-module(amoc_config_attributes_SUITE).
+
+-include_lib("eunit/include/eunit.hrl").
+-include("../src/amoc_config/amoc_config.hrl").
+
+-export([all/0]).
+-export([get_module_attributes/1,
+         get_module_configuration/1,
+         errors_reporting/1,
+         multiple_exports_error/1,
+         old_attributes_format/1,
+         one_of_function/1]).
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% these attributes and functions are required for the testing purposes %%
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+-required_variable(#{name => var0, description => "var0"}).
+-required_variable(#{name => var1, description => "var1", value => def1}).
+-required_variable([
+    #{name => var2, description => "var2", value => def2, verification => none},
+    #{name => var3, description => "var3", value => def3,
+      verification => [def3, another_atom]},
+    #{name => var4, description => "var4", value => def4, verification => is_atom},
+    #{name => var5, description => "var5", value => def5,
+      verification => fun ?MODULE:is_atom/1}
+]).
+-required_variable(#{name => var6, description => "var6", value => "def6",
+                     verification => is_list, update => read_only}).
+-required_variable(#{name => var7, description => "var7", value => def7,
+                     verification => none, update => none}).
+-required_variable([
+    #{name => var8, description => "var8", value => def8,
+      verification => none, update => update_value},
+    #{name => var9, description => "var9", value => def9,
+      verification => none, update => fun ?MODULE:update_value/2}
+]).
+
+%% verification functions
+-export([is_atom/1, none/1]).
+
+%% update functions
+-export([update_value/2, none/2]).
+
+is_atom(_) -> true.
+none(_) -> true.
+
+update_value(_, _) -> ok.
+none(_, _) -> ok.
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+all() ->
+    [get_module_attributes,
+     get_module_configuration,
+     errors_reporting,
+     multiple_exports_error,
+     old_attributes_format,
+     one_of_function].
+
```

Comment:
I think the test case names could be a bit improved to say in short what a test case does. For instance, I'm not sure what exactly does `multiple_exprts_error`. More concreate example the name `old_attributes_format` could be renamed to `old_attributes_format_is_supported`.